src/google/protobuf/stubs/platform_macros.h

```diff
@@ -115,6 +115,8 @@ GOOGLE_PROTOBUF_PLATFORM_ERROR
 // TLS support on android.
 // iOS also does not support the __thread keyword.
 #define GOOGLE_PROTOBUF_NO_THREADLOCAL
+
+#include <pthread.h>
```

Comment:
Could you move this include to where the GOOGLE_PROTOBUF_NO_THREADLOCAL macro is used? I think most files including platform_macros.h does not need pthread.h.